benches/codec.rs

```diff
@@ -0,0 +1,110 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#![feature(test)]
+extern crate parquet;
+#[macro_use]
+extern crate lazy_static;
+extern crate test;
+use test::Bencher;
+
+use std::env;
+use std::fs::File;
+
+use parquet::basic::Compression;
+use parquet::file::reader::*;
+use parquet::compression::*;
+
+// 10k rows with types:
+//   binary, int32, int64, int96, boolean, float, double, and fixed_len_byte_array
+// filled with random values.
+const TEST_FILE: &str = "10k.parquet";
```

Comment:
Could we have a benchmark that is similar to unit tests, where we test with byte arrays?

Reply:
Do you mean generate the byte arrays for different types in the benchmark (instead of reading from the Parquet file)?

Reply:
If it is possible in this case, yes. Though it looks like benchmark just compresses/decompresses byte arrays, we could make it type specific, similar to encoding/decoding tests. 

Again, only if it is possible and does not affect the benchmark.

Reply:
OK. Let me change it to be type-specific, that is, test column by column. I'd prefer to keep the Parquet file though since it allows us to test compression on different encodings. For instance, it uses PLAIN encoding for float/double, and DELTA_BINARY_PACKING for int32, etc. We can also use it to test data page v1 and v2. The file is generated by a modified version of [DataGenerator](https://github.com/apache/parquet-mr/blob/master/parquet-benchmarks/src/main/java/org/apache/parquet/benchmarks/DataGenerator.java).

Reply:
Makes sense to keep the file then. All good.